quipucords/scanner/network/processing/process.py

```diff
@@ -29,6 +31,17 @@
 SUDO_ERROR = "sudo: a password is required"
 
 
+@dataclass
+class ProcessedResult:
+    """Processed result including relevant metadata."""
+
+    # return_code == 0 = success
+    return_code: int
+    error_msg: str = None
+    # the actual fact value
+    value: Any = None
```

Comment:
This piece is more important than the feature itself, IMO.

I've been thinking how we could make the raw facts report more transparent about what happened - and help us make decisions later on how to proper use it during fingerprint/normalization and came up with this.
return code and error message are the only things that came to mind (trying to follow the same pattern that we did on openshift implementation). With this we gain granularity of what exactly happened and maybe what caused a fact to not be collected / processed.